lodash.js

```diff
@@ -4656,6 +4658,23 @@
       return object.index - other.index;
     }
 
+    /**
+     * Used by `compareMultiple` to evaluate a criterion, either reusing its
+     * computed result or computing it for the first time and then caching the
+     * result.
+     *
+     * @private
+     * @param {Object} criterion The criterion to evaluate
+     * @return {*} The comparison key
+     */
+    function forceCriterion(criterion) {
+      if (!criterion.computed) {
+        criterion.value = criterion.iteratee(criterion.element);
```

Comment:
Can you pluck the `criterion.iteratee` so that it's `this` is not associated with `criterion`:
```js
const { iteratee } = criterion

criterion.value = iteratee(criterion.element)
```